File: 251120/최대공약수 구하기/find-the-greatest-common-divisor.js
const fs = require("fs");
const input = fs.readFileSync(0).toString().trim().split(" ");
let n = Number(input[0]); // 첫 번째 정수 n
let m = Number(input[1]); // 두 번째 정수 m

// Please Write your code here.

function getGCD(num1, num2) {
    let a = num1;
    let b = num2;

    while (b !== 0) {
        const remainder = a % b;
        a = b;
        b = remainder;
    }

    return a;
}

const result = getGCD(n, m);

console.log(result);

File: 250327/최대공약수 구하기/find-the-greatest-common-divisor.java
import java.util.Scanner;

public class Main {
    public static int gcd(int a, int b) {
        if(b == 0)
            return a;
        else {
            return gcd(b, a % b);
        }

    }
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int m = sc.nextInt();
        // Please write your code here.
        System.out.print(gcd(n,m));
    }
}